src/openklant/components/klantinteracties/api/viewsets/klantcontacten.py

```diff
@@ -378,22 +380,76 @@ def perform_update(self, serializer):
         )
 
     @transaction.atomic
-    def perform_destroy(self, instance):
-        super().perform_destroy(instance)
-        token_auth = self.request.auth
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        cascade = request.query_params.get("cascade", "false").lower() == "true"
+
+        token_auth = request.auth
+        remaining_klantcontacts = []
+
+        if cascade:
+            for kc in (instance.klantcontact, instance.was_klantcontact):
+                if not kc:
+                    continue
+
+                has_other_connection = (
+                    Onderwerpobject.objects.filter(
+                        models.Q(klantcontact=kc) | models.Q(was_klantcontact=kc)
+                    )
+                    .exclude(pk=instance.pk)
+                    .exists()
+                )
+
+                if has_other_connection:
+                    continue
+
+                betrokkenen = Betrokkene.objects.filter(klantcontact=kc)
+                for betrokkene in betrokkenen:
+                    DigitaalAdres.objects.filter(
+                        betrokkene=betrokkene,
+                        partij__isnull=True,
+                    ).delete()
+
+                kc.delete()
+
+            instance.delete()
+
+            logger.info(
+                "onderwerpobject_deleted_cascade",
+                uuid=str(instance.uuid),
+                token_identifier=token_auth.identifier,
+                token_application=token_auth.application,
+            )
+
+            return Response(status=status.HTTP_204_NO_CONTENT)
+
+        if instance.klantcontact:
+            remaining_klantcontacts.append(
+                request.build_absolute_uri(
+                    reverse(
+                        "klantinteracties:klantcontact-detail",
+                        kwargs={
+                            "uuid": instance.klantcontact.uuid,
+                            "version": self.request.version or "1",
+                        },
+                    )
+                )
+            )
+
+        instance.delete()
+
         logger.info(
-            "onderwerpobject_deleted",
+            "onderwerpobject_deleted_no_cascade",
             uuid=str(instance.uuid),
             klantcontact_uuid=str(instance.klantcontact.uuid)
             if instance.klantcontact
             else None,
-            was_klantcontact_uuid=str(instance.was_klantcontact.uuid)
-            if instance.was_klantcontact
-            else None,
             token_identifier=token_auth.identifier,
             token_application=token_auth.application,
         )
 
+        return Response(remaining_klantcontacts, status=status.HTTP_200_OK)
```

Comment:
Could you also make sure this response is documented in the API spec? You can also do this with `extend_schema` on the viewset

```python
extend_schema(responses={
            status.HTTP_204_NO_CONTENT: None,
            status.HTTP_200_OK: ...
            }
)
```